cmd/gardenlet/app/app.go

```diff
@@ -451,6 +465,139 @@ func (g *garden) Start(ctx context.Context) error {
+				if !apierrors.IsNotFound(err) {
+					return err
+				}
+			}
+			if err := seedClient.Delete(ctx, &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: dwd.ExternalProbeSecretName, Namespace: namespace.Name}}); err != nil {
+				if !apierrors.IsNotFound(err) {
+					return err
+				}
+			}
+
+			// Fetch and update the GRM configmap
+			grmConfigMap := &corev1.ConfigMap{}
+			var grmCMName string
+			var grmCMVolumeIndex int
+			for n, vol := range grmDeploy.Spec.Template.Spec.Volumes {
+				if vol.Name == "config" {
+					grmCMName = vol.ConfigMap.Name
+					grmCMVolumeIndex = n
+					break
+				}
+			}
+			if len(grmCMName) == 0 {
+				return nil
+			}
+			if err := seedClient.Get(ctx, types.NamespacedName{Namespace: namespace.Name, Name: grmCMName}, grmConfigMap); err != nil {
+				if apierrors.IsNotFound(err) {
+					return nil
+				}
+				return err
+			}
+
+			cmData := grmConfigMap.Data["config.yaml"]
+			rmConfig := resourcemanagerv1alpha1.ResourceManagerConfiguration{}
+
+			// create codec
+			var codec runtime.Codec
+			configScheme := runtime.NewScheme()
+			utilruntime.Must(resourcemanagerv1alpha1.AddToScheme(configScheme))
+			utilruntime.Must(apiextensionsv1.AddToScheme(configScheme))
+			ser := json.NewSerializerWithOptions(json.DefaultMetaFactory, configScheme, configScheme, json.SerializerOptions{
+				Yaml:   true,
+				Pretty: false,
+				Strict: false,
+			})
+			versions := schema.GroupVersions([]schema.GroupVersion{
+				resourcemanagerv1alpha1.SchemeGroupVersion,
+				apiextensionsv1.SchemeGroupVersion,
+			})
+			codec = serializer.NewCodecFactory(configScheme).CodecForVersions(ser, ser, versions, versions)
+
+			obj, err := runtime.Decode(codec, []byte(cmData))
+			if err != nil {
+				return err
+			}
+			rmConfig = *(obj.(*resourcemanagerv1alpha1.ResourceManagerConfiguration))
+
+			rmConfig.TargetClientConnection.Namespaces = append(rmConfig.TargetClientConnection.Namespaces, corev1.NamespaceNodeLease)
+
+			data, err := runtime.Encode(codec, &rmConfig)
+			if err != nil {
+				return err
+			}
+
+			if err = seedClient.Delete(ctx, grmConfigMap); err != nil {
+				return err
+			}
+
+			newGRMConfigMap := &corev1.ConfigMap{}
+			newGRMConfigMap.Data = map[string]string{"config.yaml": string(data)}
+			newGRMConfigMap.Immutable = ptr.To[bool](true)
+			newGRMConfigMap.ObjectMeta.Name = dwd.TempGRMConfigMapName
+			newGRMConfigMap.ObjectMeta.Namespace = namespace.Name
+			newGRMConfigMap.ObjectMeta.Labels = grmConfigMap.ObjectMeta.Labels
+
+			if err = seedClient.Create(ctx, newGRMConfigMap); err != nil {
+				if !apierrors.IsAlreadyExists(err) {
+					return err
+				}
+			}
+
+			patch := client.MergeFrom(grmDeploy.DeepCopy())
+			grmDeploy.Spec.Template.Spec.Volumes[grmCMVolumeIndex].ConfigMap.Name = dwd.TempGRMConfigMapName
+
+			return seedClient.Patch(ctx, grmDeploy, patch)
+		})
```

Comment:
We should do this only if `TargetClientConnection.Namespaces` doesn't have NamespaceNodeLease.